### PR TITLE
Support absolute paths in asset distribution definition

### LINKED ISSUE #218: Support absolute paths when defining assets in sls-asset-distribution
Often the path used comes from gradle, e.g. `${buildDir}/something` which gives you an absolute path. Doing this results in an empty bundle however with no warning that the contents weren't copied.

### PR DESCRIPTION
**Pull request by gatesn**
Fixes #218

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/sls-packaging/219)

### CODE REVIEW

File: gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/asset/tasks/AssetDistTarTask.groovy [hunk 1]
```diff
@@ -36,7 +36,7 @@ class AssetDistTarTask {
             }
 
             assetDirs.entrySet().each { entry ->
-                from("${project.projectDir}/${entry.getKey()}") {
+                from(file(project.projectDir, entry.getKey())) {
```
Comment (uschi2000) [inline gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/asset/tasks/AssetDistTarTask.groovy]: what is this `file()` method?

**Comment by gatesn** [comment]
It's a gradle builtin that basically wraps `new File(...)`

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by uschi2000** [comment]
That's a method on the `Project` class.

**Comment by uschi2000** [comment]
(just appears magically as `file()` in build.gradle scripts)

**Comment by gatesn** [comment]
And does it resolve against project root by default?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by uschi2000** [comment]
yes.

**Comment by gatesn** [comment]
Changed to just `file(entry.getKey())`

File: gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/asset/tasks/AssetDistTarTask.groovy [hunk 2]
```diff
@@ -36,7 +36,7 @@ class AssetDistTarTask {
             }
 
             assetDirs.entrySet().each { entry ->
-                from("${project.projectDir}/${entry.getKey()}") {
+                from(file(entry.getKey())) {
```
Comment (uschi2000) [inline gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/asset/tasks/AssetDistTarTask.groovy]: it's `project.file()`

**Review by quellkot — APPROVED** [review]

**Review by uschi2000 — APPROVED** [review]

**Pull request merged**